### PR TITLE
Add multiple product types filter to products

### LINKED ISSUE #5172: Add multiple product types filter to products
### What I'm trying to achieve
To be able to filter products by multiple types.

### Describe a proposed solution
```
input ProductFilterInput {
  isPublished: Boolean
  collections: [ID]
  categories: [ID]
  hasCategory: Boolean
  price: PriceRangeInput
  attributes: [AttributeInput]
  stockAvailability: StockAvailability
+ productType: ID @deprecated
+ productTypes: [ID]
  search: String
  minimalPrice: PriceRangeInput
}
```

It would also require to make this change
```
input ProductTypeFilterInput {
  search: String
  configurable: ProductTypeConfigurable
  productType: ProductTypeEnum
+ ids: [ID]
}
```

### PR DESCRIPTION
**Pull request by kswiatek92**
closes #5172 

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 8c51fab5fa7bf0532dd130f14dc7b08327e1127f (kswiatek92/saleor @ add-multiple-product-types-filter-to-products)
Base comparison is 77c54b798d0d8b70cde3dfab23e3643acb51837e.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         10	         10	              0
  complete checkout                          	          8	          8	              0
  create checkout                            	          5	          5	              1

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         18	         18	              4
  retrieve product attributes                	          9	          9	              0

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         23	         23	              9

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/graphql/product/filters.py [hunk 1]
```diff
@@ -2,9 +2,11 @@
 from typing import Dict, List, Optional
 
 import django_filters
+from django.core.exceptions import ValidationError
```
Comment (maarcingebala) [inline saleor/graphql/product/filters.py]: Is it used?

File: saleor/graphql/product/filters.py [hunk 2]
```diff
@@ -2,9 +2,11 @@
...
 from django.db.models import Q, Subquery, Sum
 from graphene_django.filter import GlobalIDFilter, GlobalIDMultipleChoiceFilter
 
+from ...product.error_codes import ProductErrorCode
```
Comment (maarcingebala) [inline saleor/graphql/product/filters.py]: And this?

File: saleor/graphql/product/types/attributes.py [hunk 3]
```diff
@@ -161,4 +161,9 @@ class Meta:
 
 class AttributeInput(graphene.InputObjectType):
     slug = graphene.String(required=True, description=AttributeDescriptions.SLUG)
-    value = graphene.String(required=True, description=AttributeValueDescriptions.SLUG)
+    value = graphene.String(
+        required=False, description=AttributeValueDescriptions.SLUG
```
Comment (maarcingebala) [inline saleor/graphql/product/types/attributes.py]: If this is deprecated we should add information in the description as well. See as it's done in other deprecated input fields.

**Pull request merged**